pkg/controller/bootstrap/bootstrap.go

```diff
@@ -272,6 +277,29 @@ func (b *Bootstrap) Run(destDir string) error {
 			return err
 		}
 	}
+
+	// If an apiServer object exists, write it to /etc/mcs/bootstrap/api-server/api-server.yaml
+	// so that bootstrap MCS can consume it
+	if apiServer != nil {
+		buf := bytes.Buffer{}
+		encoder := codecFactory.EncoderForVersion(serializer, apicfgv1.GroupVersion)
+		err = encoder.Encode(apiServer, &buf)
+		if err != nil {
+			return err
+		}
+		apiserverDir := filepath.Join(destDir, "api-server")
+		if err := os.MkdirAll(apiserverDir, 0o764); err != nil {
```

Comment:
question: Hmm, I see we have this permission for some files, but I'm not really sure why we need execute and group write, etc.? Non-blocking though

Reply:
I looked up the history on this and found these [[1]](https://github.com/openshift/machine-config-operator/pull/32#discussion_r213470034) and [[2]](https://github.com/djoshy/machine-config-operator/commit/eeb1fad5f1fdf80b410657f2f5538c0806c2d0f9). The folders get the owner execute bit, but the files don't. But, definitely agree that some updates could be done here. Perhaps a tech debt card?